### ISSUE TITLE
multiplayer locked 1.16.5 - 1.16.4

### ISSUE THREAD

**Issue by kolyax** [elyby/accounts#38]
```
from turtle import width
import PIL
import PIL.Image
import minecraft_launcher_lib
import os
import sys
import tkinter
import tkinter as tk
from tkinter import ttk
from tkinter.messagebox import showinfo
import minecraft_launcher_lib.utils
import minecraft_launcher_lib
import subprocess
import requests
import shutil
from bs4 import BeautifulSoup
import os
import sys
from PIL import Image, ImageTk


# create the root window
def resource_path(relative_path):
    """Get absolute path to resource, works for dev and for PyInstaller"""
    try:
        # PyInstaller creates a temp folder and stores path in _MEIPASS
        base_path = sys._MEIPASS
    except Exception:
        base_path = os.path.abspath(".")

    return os.path.join(base_path, relative_path)


def download_with_redirect(url, save_path):
    # Отправляем запрос и разрешаем редиректы
    response = requests.get(url, allow_redirects=True)

    # Проверяем успешный ответ
    if response.status_code == 200:
        # Сохраняем файл
        with open(save_path, "wb") as file:
            file.write(response.content)
        print(f"Файл успешно загружен: {save_path}")
    else:
        print(f"Ошибка загрузки, код ответа: {response.status_code}")


def get_forge_url(mcversion, version, branch=None, category="installer", format1="jar"):
    # Формируем URL для скачивания
    branch_part = f"-{branch}" if branch else ""
    url = f"https://files.minecraftforge.net/maven/net/minecraftforge/forge/{mcversion}-{version}{branch_part}/forge-{mcversion}-{version}{branch_part}-{category}.{format1}"
    return url


# Пример использования:
def getforgeversion(mcversion):
    response = requests.get(
        f"https://bmclapi2.bangbang93.com/forge/minecraft/{mcversion}",
        allow_redirects=True,
    )
    print(response.text)
    jsonl = json.loads(response.text)
    corect = "0000"
    corectv = ""
    for i in jsonl:
        try:
            for j in i["files"]:
                if j["format"] == "jar":
                    print(1)
                    if j["category"] == "installer":
                        print(2)
                        if int(i["version"].replace(".", "")) > int(corect):
                            print(3)
                            corectv = i["version"]
                            corect = int(i["version"].replace(".", ""))
        except Exception as e:
            print(e)

    return corectv


def downloadforge():
    mcversion = input("version:")
    # Формируем ссылку на файл
    g = getforgeversion(mcversion)
    download_url = get_forge_url(mcversion, g, None, "installer", "jar")

    # Путь для сохранения файла
    save_path = os.path.join("downloads", f"forge-{mcversion}-{g}.jar")
    os.makedirs(os.path.dirname(save_path), exist_ok=True)

    # Скачиваем файл
    download_with_redirect(download_url, save_path)
    subprocess.call(
        f"java -jar {save_path} --installClient C:\\Users\\abobu\\AppData\\Roaming\\.chto_za_zalupa",
        shell=False,
    )


def start_or_install(a1, a2):
    window.destroy()

    ver = a1
    user = a2

    def delete_folder_with_contents(folder_path):
        # Проверяем, существует ли папка
        if os.path.exists(folder_path):
            # Удаляем папку и все её содержимое
            shutil.rmtree(folder_path)
            print(f"Папка '{folder_path}' и её содержимое успешно удалены.")
        else:
            print(f"Папка '{folder_path}' не существует.")

    def copy_directory(source, destination):
        # Проверяем, существует ли исходная папка
        if not os.path.exists(source):
            print(f"Исходная папка '{source}' не существует.")
            return

        # Проверяем, существует ли папка назначения
        if os.path.exists(destination):
            print(
                f"Папка назначения '{destination}' уже существует. Удалите её или выберите другую."
            )
            return

        # Копирование папки со всем её содержимым
        shutil.copytree(source, destination)
        print(f"Папка '{source}' успешно скопирована в '{destination}'.")

    minecraft_directory = (
        minecraft_launcher_lib.utils.get_minecraft_directory().replace(
            "minecraft", "chto_za_zalupa"
        )
    )

    def printProgressBar(
        iteration,
        total,
        prefix="",
        suffix="",
        decimals=1,
        length=100,
        fill="█",
        printEnd="\r",
    ):
        percent = ("{0:." + str(decimals) + "f}").format(
            100 * (iteration / float(total))
        )
        filledLength = int(length * iteration // total)
        bar = fill * filledLength + "-" * (length - filledLength)
        print("\r%s |%s| %s%% %s" % (prefix, bar, percent, suffix), end=printEnd)
        if iteration == total:
            print()

    def maximum(max_value, value):
        max_value[0] = value

    version = ver

    print(
        "======================================================================================="
    )

    max_value = [0]

    callback = {
        "setStatus": lambda text: print(text, end="r"),
        "setProgress": lambda value: printProgressBar(value, max_value[0]),
        "setMax": lambda value: maximum(max_value, value),
    }

    # Install Minecraft version
    minecraft_launcher_lib.install.install_minecraft_version(
        versionid=version, minecraft_directory=minecraft_directory, callback=callback
    )
    minecraft_directory = (
        minecraft_launcher_lib.utils.get_minecraft_directory().replace(
            "minecraft", "chto_za_zalupa"
        )
    )
    folder_to_delete = minecraft_directory + "/libraries/com/mojang/authlib"
    delete_folder_with_contents(folder_to_delete)
    copy_directory(
        resource_path("authlib"), minecraft_directory + "/libraries/com/mojang/authlib"
    )

    options = {
        "username": user,
        "offline-mode": True,
    }
    d = [
        "Dminecraft.api.auth.host",
        "Dminecraft.api.account.host",
        "Dminecraft.api.session.host",
        "Dminecraft.api.services.host",
    ]
    dd = []
    f = 0
    for i in minecraft_launcher_lib.command.get_minecraft_command(
        version=version, minecraft_directory=minecraft_directory, options=options
    ):
        if not i in d:
            if f == 1:
                # dd.append(f"-javaagent:C:/Users/abobu\AppData/Roaming/.chto_za_zalupA/authlib-injector.jar=accountsHost=account.ely.by")
                pass
            dd.append(i)
            f += 1
    print(dd)
    input()
    subprocess.call(dd)
    openl()


def load_asset(path):
    base = getattr(sys, "_MEIPASS", os.path.dirname(os.path.abspath(__file__)))
    assets = os.path.join(base, "assets")
    return os.path.join(assets, path)


def openl():
    global window, text1
    window = tkinter.Tk()
    window.geometry("1600x800")
    window.configure(bg="#ffffff")
    window.title("Untitled")

    canvas = tkinter.Canvas(
        window,
        bg="#ffffff",
        width=1600,
        height=800,
        bd=0,
        highlightthickness=0,
        relief="ridge",
    )

    canvas.place(x=0, y=0)

    image_1 = tkinter.PhotoImage(file=resource_path("assets/1.png"))

    canvas.create_image(1600 / 2, 800 / 2, image=image_1)

    button_1_image = ImageTk.PhotoImage(
        Image.open(resource_path("assets/2.png")).convert("RGBA")
    )

    button_1 = tk.Button(
        image=button_1_image,
        relief="flat",
        borderwidth=0,
        highlightthickness=0,
        command=lambda: start_or_install(text1, "kolyax"),
    )
    button_1.place(x=32, y=0, width=219, height=69)

    canvas.create_line(32, 272, 251, 272, fill="#000000", width=1.0)

    canvas.create_text(
        32, 273, anchor="nw", text="version", fill="#ffffff", font=("Inter", 20 * -1)
    )

    text1 = canvas.create_text(
        32, 249, anchor="nw", text="version", fill="#ffffff", font=("Inter", 20 * -1)
    )

    image_2 = tkinter.PhotoImage(file=resource_path("assets/3.png"))

    canvas.create_image(1463, 59, image=image_2)

    image_3 = tkinter.PhotoImage(file=resource_path("assets/4.png"))

    canvas.create_image(1462, 203, image=image_3)

    image_4 = tkinter.PhotoImage(file=resource_path("assets/5.png"))

    canvas.create_image(1462, 374, image=image_4)

    canvas.create_line(1201, 131, 1496, 131, fill="#000000", width=1.0)

    canvas.create_line(1201, 272, 1496, 272, fill="#000000", width=1.0)

    canvas.create_line(1201, 434, 1496, 434, fill="#000000", width=1.0)

    canvas.create_text(
        1214,
        141,
        anchor="nw",
        text="nikname of user",
        fill="#ffffff",
        font=("Inter", 20 * -1),
    )

    canvas.create_text(
        1214,
        282,
        anchor="nw",
        text="to registration clikc to gear",
        fill="#ffffff",
        font=("Inter", 20 * -1),
    )

    canvas.create_text(
        1214, 434, anchor="nw", text="website", fill="#ffffff", font=("Inter", 20 * -1)
    )
    text1 = "1.18.2"

    def items_selected(event):
        global text1
        # get selected indices
        selected_indices = listbox.curselection()
        # get selected items
        selected_langs = ",".join([listbox.get(i) for i in selected_indices])
        msg = f"You selected: {selected_langs}"
        text1 = selected_langs

    canvas.create_line(1201, 131, 1496, 131, fill="#000000", width=1.0)
    d = []
    minecraft_directory = (
        minecraft_launcher_lib.utils.get_minecraft_directory().replace(
            "minecraft", "chto_za_zalupa"
        )
    )
    for i in minecraft_launcher_lib.utils.get_version_list():
        if i["type"] == "release":
            if True:
                d.append(i["id"])

    # r = requests.get("https://bmclapi2.bangbang93.com/forge/minecraft").text
    # listf = r.replace("[","").replace("]","").replace('"',"").split(",")
    # for i in listf:
    #    d.append("Forge "+i)
    var = tkinter.Variable(value=d)

    listbox = tk.Listbox(window, listvariable=var, height=9, selectmode=tk.ACTIVE)

    listbox.place(
        x=32,
        y=80,
    )
    listbox.bind("<<ListboxSelect>>", items_selected)
    # link a scrollbar to a list
    scrollbar = ttk.Scrollbar(window, orient=tkinter.VERTICAL, command=listbox.yview)

    listbox["yscrollcommand"] = scrollbar.set

    scrollbar.place(x=32, y=80, height=0)
    window.resizable(False, False)
    window.mainloop()


openl()
```
im can't solve it 8 days
![Screenshot 2024-12-21 132627](https://github.com/user-attachments/assets/78fc7fff-3a38-4e2c-a5aa-e815d51d26a2)

**Comment by kolyax** [comment]
folder authlib 
[authlib.zip](https://github.com/user-attachments/files/18218348/authlib.zip)

**Comment by kolyax** [comment]
in other version all work

**Comment by Octol1ttle** [comment]
Thanks for reporting this issue. It has been fixed in versions `2.1.28-ely.3` and `2.3.31-ely.3` (it may take a while before they appear in the Maven). To work around this issue on `-ely.2`, set the `ely.authlib.skipFetchPrivileges` system property to `true`.

**Comment by erickskrauch** [comment]
Thank you for reporting this bug. @Octol1ttle fixed it and deployed it to the Ely.by's [load page](https://ely.by/load) and Legacy Launcher's repositories.
